lib/ui/geometry.dart

```diff
@@ -12,7 +12,11 @@ abstract class OffsetBase {
   ///
   /// The first argument sets the horizontal component, and the second the
   /// vertical component.
-  const OffsetBase(this._dx, this._dy);
+  const OffsetBase(double dx, double dy)
+      : _dx = dx ?? 0.0
+      , _dy = dy ?? 0.0
+      , assert(dx != null)
+      , assert(dy != null);
```

Comment:
nit: can this be made consistent with the rest of the file? I think commas go at the end of the lines rather than the start.

Reply:
Is it enough to just assert these are not null rather than coalescing?  Here and below.

Reply:
I would rather a user know they tried to pass null where they shouldn't than find out they passed null and got 0.

It's already an error to pass null based on the rest of the code.

Reply:
Unfortunately we can't just add an assert, because the compiler needs to be able to prove that the field will never be `null` in release mode, where the asserts aren't triggered.

If the constructor wasn't const, we would throw an exception unconditionally instead. Is it not sufficient to prevent those kinds of errors with an assert in debug mode?

Reply:
Ahh that makes sense.

I think this is fine then.  Thanks!